File: Servidor/grapeguardapi/src/main/java/com/raquelbytes/grapeguardapi/Model/Cosecha.java
package com.raquelbytes.grapeguardapi.Model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Data
@Table(name = "Cosecha")
public class Cosecha {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID_cosecha")
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "ID_vinedo", nullable = false)
    @JsonIgnore
    private Vinedo vinedo;

    @Column(name = "nombre_variedad", nullable = false)
    private String nombreVariedad;

    @Column(name = "cantidad_uvas", nullable = false)
    private BigDecimal cantidadUvas;

    @Column(name = "precio_venta_kg", nullable = false)
    private BigDecimal precioVentaKg;

    @Column(name = "fecha_cosecha", nullable = false)
    private LocalDate fechaCosecha;


}

File: Movil/GrapeGuard/app/src/main/java/com/raquelbytes/grapeguard/API/Model/Cosecha.kt
package com.raquelbytes.grapeguard.API.Model

import java.math.BigDecimal
import java.time.LocalDate

 class Cosecha{
    var id: Int? = null
    var vinedo: Vinedo? = null
    var nombreVariedad: String? = null
    var cantidadUvas: BigDecimal? = null
    var precioVentaKg: BigDecimal? = null
    var fechaCosecha: String? = null

constructor()

constructor(id: Int, vinedo: Vinedo, nombreVariedad: String, cantidadUvas: BigDecimal, precioVentaKg: BigDecimal, fechaCosecha: String) {
    this.id = id
    this.vinedo = vinedo
    this.nombreVariedad = nombreVariedad
    this.cantidadUvas = cantidadUvas
    this.precioVentaKg = precioVentaKg
    this.fechaCosecha = fechaCosecha
}
}
